### PR TITLE
Feature/clean up proto abstractions

### PR DESCRIPTION
**Pull request by davidfowl**
This is a do-over of https://github.com/aspnet/KestrelHttpServer/pull/2381. I need to react in SignalR before I merge this.

Blocked by https://github.com/aspnet/SignalR/pull/1583

### CODE REVIEW

File: src/Kestrel.Transport.Abstractions/Internal/TransportConnection.Features.cs [hunk 1]
```diff
@@ -160,18 +199,30 @@ IDuplexPipe IConnectionTransportFeature.Application
 
         TFeature IFeatureCollection.Get<TFeature>()
         {
-            if (typeof(TFeature) == typeof(IHttpConnectionFeature))
+            if (typeof(TFeature) == IHttpConnectionFeatureType)
```
Comment (benaadams) [inline src/Kestrel.Transport.Abstractions/Internal/TransportConnection.Features.cs]: The generic version should use `typeof` for the Jit so its `T` == `T`
`typeof(TFeature) == typeof(IHttpConnectionFeature)`

Comment (benaadams) [reply src/Kestrel.Transport.Abstractions/Internal/TransportConnection.Features.cs]: `Type` version is different as its `variable == `

Comment (davidfowl) [reply src/Kestrel.Transport.Abstractions/Internal/TransportConnection.Features.cs]: Good to know

Comment (davidfowl) [reply src/Kestrel.Transport.Abstractions/Internal/TransportConnection.Features.cs]: I'll fix this in a subsequent commit.

**Pull request merged**